docs/code-examples/line_strip3d_simple.cpp

```diff
@@ -17,6 +17,6 @@ int main() {
         {1.f, 1.f, 1.f},
         {0.f, 1.f, 0.f},
         {0.f, 1.f, 1.f},
-    };
-    rec.log("strip", rr::LineStrips3D(points));
+    });
+    rec.log("strip", rr::LineStrips3D(linestrip));
```

Comment:
Am I reading this right: a single strip is passed to a constructor that can _also_ take multiple strips?

Reply:
correct!

Reply:
Is this implemented with an overloaded constructor to ComponentBatch? There's so many layers to this 😓 implicit constructor calling in C++ is a bit 😬

Reply:
This one is _relatively_ straight forward :)

```
    /// Adapter for a single component, temporary or reference.
    template <typename TComponent>
    struct ComponentBatchAdapter<TComponent, TComponent> {
        ComponentBatch<TComponent> operator()(const TComponent& one_and_only) {
            return ComponentBatch<TComponent>::borrow(&one_and_only, 1);
        }

        ComponentBatch<TComponent> operator()(TComponent&& one_and_only) {
            return ComponentBatch<TComponent>::take_ownership(std::move(one_and_only));
        }
    };
```